homeassistant/components/ambient_station/__init__.py

```diff
@@ -286,6 +286,36 @@
     return True
 
 
+async def async_migrate_entry(hass, config_entry):
+    """Migrate old entry."""
+    version = config_entry.version
+
+    _LOGGER.debug('Migrating from version %s', version)
+
+    reasons_to_re_add = {
+        1: "The integration's unique ID format has changed"
+    }
+
+    # 1 -> 2: Unique ID format changed, so delete and re-import:
```

Comment:
Why do we need to re-import? Why not just go into to dev and ent registries and delete it all ? 

This seems very time sensitive, because we are continuing to set up the config entry, only to remove it, although remove might hit before set up is done ?

Reply:
Sorry, not following. Are you saying I should just delete all related entries in the device and entity registries, then allow startup to continue as normal?

Reply:
I think so. I am afraid that any change to our logic that makes either remove or setup shorter/longer will cause this to fail.

Reply:
48f0ec1?